benchmark/single-source/BucketSort.swift

```diff
@@ -0,0 +1,139 @@
+//===--- BucketSort.swift ----------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+// Adapted from the original implementation: 
+// https://github.com/raywenderlich/swift-algorithm-club/tree/master/Bucket%20Sort
+// Issue: https://github.com/raywenderlich/swift-algorithm-club/issues/863
+
+// This benchmark implements the classical BucketSort algorithm described at
+// https://en.wikipedia.org/wiki/Bucket_sort. The implementation allows for an
+// array of generic type of "SortableItem" to be sorted. Unfortunately,  if
+// ``sortingAlgo'' type is not known for the callsite in line 84, then the 
+// ``sort'' method can not be specialized to integer array sorting, which will 
+// lead to a huge performance loss. Since SortingAlgorithm and InsertionSort are 
+// declared to be ``public'' and that lines 83-85 can not be inlined in
+// BucketSortImpl (due to inlining heuristic limitations), today swift 
+// compiler (without ExistentialSpecializer) can not achieve this feat.  With 
+// ExistentialSpecializer which enables generic specialization recursively in a
+// call chain, we are able to specialize line 84 for InsertionSort on integers.
+
+import TestsUtils
+import Foundation
+
+public let BucketSort = BenchmarkInfo(
+  name: "BucketSort",
+  runFunction: run_BucketSort,
+  tags: [.validation, .algorithm],
+  setUpFunction: { buildWorkload() },
+  legacyFactor: 10)
```

Comment:
@rajbarik apparently `legacyFactor` is not something you're supposed to define. You can just remove it.

@eeckstein @gottesmm Please comment the purpose of this setting in TestsUtils.swift.

Reply:
Sorry for causing confusion, I’ll fix the docs tomorrow.